source/common/http/conn_pool_grid.cc

```diff
@@ -205,17 +236,22 @@ ConnectionPool::Cancellable* ConnectivityGrid::newStream(Http::ResponseDecoder&
   if (pools_.empty()) {
     createNextPool();
   }
-
-  // TODO(#15649) track pools with successful connections: don't always start at
-  // the front of the list.
-  auto wrapped_callback =
-      std::make_unique<WrapperCallbacks>(*this, decoder, pools_.begin(), callbacks);
+  PoolIterator pool = pools_.begin();
+  if (is_http3_broken_) {
+    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", describePool(**pool),
+              host_->hostname());
+    // Since HTTP/3 is broken, presumably both pools have already been created so this
+    // is just to be safe.
+    createNextPool();
+    pool = ++pool;
```

Comment:
nit: this is odd, you're incrementing the variable then assigning it to itself? This smells very close to undefined behavior (even though I don't think it is). How about this instead? `pool = std::next(pool);`

Reply:
Oh how funny! Not sure how I did that. In any case simply "++pool" will be fine, which I've now done.